Automattic/Gravatar-SDK-Android
# QuickEditor: Avatar Rating - part 1

Part of #237

### Description

This PR adds an option for a nested Popup Menu to accommodate the requirement for Avatar rating. 

To achieve that I modified the `PickerPopupItem` to contain sub-elements and wrapped the list of those in a `PickerPopupMenu` object. This is then used in the `PickerPopup` to switch the state when an item with submenu is clicked. For the animation, I used `AnimatedContent` component.

This PR doesn't implement the full functionality, only the UI part of the popup menu. I will add all the API requests in the following one. 

I've also skipped any extra UI improvements like arrows when showing the nested menu for now. We can iterate over later.


https://github.com/user-attachments/assets/4bfb2ad0-ed6b-4baf-be30-eb45421805cf



### Testing Steps

1. Open the QE
2. Tap the `...` button on an Avatar
3. Tap `Change rating`
4. Confirm the rating options are visible
5. Tap on one
6. Confirm the Popup was dismissed
7. Tap the `...` button on an Avatar
8. Tap `Change rating`
9. Tap outside of the Popup
10. Confirm the Popup was dismissed
